File: src/app/Class/UserAlbumStatus.ts
import {Album} from './Album';
import {Users} from './Users';

export class UserAlbumStatus {
  id: number;
  album: Album;
  user: Users;
  owned: boolean;
  sought: boolean;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.album = (obj && obj.album) || null;
    this.user = (obj && obj.user) || null;
    this.owned = (obj && obj.owned) || false;
    this.sought = (obj && obj.sought) || false;
  }
}

File: Backend/src/main/java/musiccube/entities/UserAlbumStatus.java
package musiccube.entities;

import javax.persistence.*;

@Entity
public class UserAlbumStatus {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Album album;
    @ManyToOne
    private User user;

    @Column
    private boolean owned;
    @Column
    private boolean sought;

    public int getId() {
        return id;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public boolean isOwned() {
        return owned;
    }

    public void setOwned(boolean owned) {
        this.owned = owned;
    }

    public boolean isSought() {
        return sought;
    }

    public void setSought(boolean sought) {
        this.sought = sought;
    }
}
